contractcourt/htlc_timeout_resolver.go

```diff
@@ -462,28 +462,14 @@ func (h *htlcTimeoutResolver) handleCommitSpend(
 			return nil, err
 		}
 
-		// Find the input index of the second-level spend, as the
-		// sweeper might have re-arranged inputs and outputs.
-		prevOutPoint := h.htlcResolution.SignedTimeoutTx.TxIn[0].PreviousOutPoint
-		ok, index := input.FindInputIndex(
-			commitSpend.SpendingTx, prevOutPoint,
-		)
-		if !ok {
-			err := fmt.Errorf("unable to find second-level "+
-				"input %v on spending tx %v", prevOutPoint,
-				commitSpend.SpenderTxHash)
-			log.Error(err)
-			return nil, err
-		}
-
 		// We'll use this input index to determine the second-level
 		// output index on the transaction, as the signatures requires
 		// the indexes to be the same. We don't look for the
 		// second-level output script directly, as there might be more
 		// than one HTLC output to the same pkScript.
 		op := &wire.OutPoint{
 			Hash:  *commitSpend.SpenderTxHash,
-			Index: index,
+			Index: commitSpend.SpenderInputIndex,
```

Comment:
doesn't the notifier dispatch on scripts rather than inputs?

Reply:
Not sure what you mean?

Reply:
before we had `FindInputIndex` which looked at the prevout to find the index of the output. The notifier dispatches on scripts, which means we'll end up in a situation where it returns the wrong index if there are duplicates, since (i believe) it'll always stop at the first.

Reply:
okay nvm! it looks like the index will be correct since we dispatch on both outpoint and script https://github.com/lightningnetwork/lnd/blob/1a73bc7d7407bcbf43886d9855dd95e747a798b0/chainntnfs/txnotifier.go#L1464L1478